protocol/shannon/config.go

```diff
@@ -65,6 +66,25 @@ type (
 		GatewayAddress          string               `yaml:"gateway_address"`
 		GatewayPrivateKeyHex    string               `yaml:"gateway_private_key_hex"`
 		OwnedAppsPrivateKeysHex []string             `yaml:"owned_apps_private_keys_hex"`
+		ServiceFallback         []ServiceFallback    `yaml:"service_fallback"`
+	}
+
+	// TODO_TECHDEBT(@adshmh): Make configuration and implementation explicit:
+	// - Criteria to decide whether the "fallback" URL should be used at all.
+	// - Criteria to decide the order in which a Shannon endpoint vs. a fallback URL should be used.
+	// - Support "weighted" distribution to Shannon endpoints vs. "fallback" URLs.
+	// - Remove the `serviceFallbackConfig` struct: update and use `ServiceFallback` as the canonical struct everywhere.
+	//
+	// ServiceFallback is a configuration struct for specifying fallback endpoints for a service.
+	ServiceFallback struct {
+		ServiceID    protocol.ServiceID `yaml:"service_id"`
+		FallbackURLs []string           `yaml:"fallback_urls"`
+
+		// TODO_TECHDEBT(@adshmh): Explicitly define and implement what happens if this is false.
```

Comment:
This is already implemented: fallback endpoints are selected by the `qos` package in the same way as any other endpoints.